### PR TITLE
Use pathtype for typed paths inside the test suites.

### PR DESCRIPTION
**Pull request by patrickt**
It’s my ultimate goal to change `Data.Project` and `Data.File` and friends to use `pathtype`, but doing such a change from the outside-in (i.e. modifying those types' consumers before the types themselves) has proven easier than doing it from the inside-out.

### CODE REVIEW

File: src/Data/Blob/IO.hs [hunk 1]
```diff
@@ -38,7 +41,11 @@ readBlobsFromDir :: MonadIO m => FilePath -> m [Blob]
 readBlobsFromDir path = liftIO . fmap catMaybes $
   findFilesInDir path supportedExts mempty >>= Async.mapConcurrently (readBlobFromFile . fileForPath)
 
--- | Read all blobs from a git repo
+  readBlobsFromGitRepoPath :: (Part.AbsRel ar, MonadIO m) => Path.Dir ar -> Git.OID -> [Path.RelFile] -> [Path.RelFile] -> m [Blob]
```
Comment (robrix) [inline src/Data/Blob/IO.hs]:
```suggestion
readBlobsFromGitRepoPath :: (Part.AbsRel ar, MonadIO m) => Path.Dir ar -> Git.OID -> [Path.RelFile] -> [Path.RelFile] -> m [Blob]
```

File: test/Examples.hs [hunk 2]
```diff
@@ -62,32 +64,33 @@ main = withOptions opts $ \ config logger statter -> hspec . parallel $ do
...
-      where stripComments = filter (\line -> not (BC.null line) && BC.head line == '#')
+      known <- BC.lines <$> B.readFile (Path.toString (tsDir </> path))
+      let stripComments = filter (\line -> not (BC.null line) && BC.head line == '#')
+      let failures = Path.relFile . BC.unpack <$> stripComments known
+      pure ((tsDir </>) <$> failures)
 
 data LanguageExample
   = LanguageExample
-  { languageName :: FilePath
-  , languageExtension :: FilePath
```
Comment (robrix) [inline test/Examples.hs]: lol

File: test/Graphing/Calls/Spec.hs [hunk 3]
```diff
@@ -6,19 +6,19 @@ import Prelude hiding (readFile)
...
 import           "semantic" Data.Graph (Graph (..), topologicalSort)
 import           Data.Graph.ControlFlowVertex
 import qualified Data.Language as Language
 import           Semantic.Graph
+import qualified System.Path as Path
 
-callGraphPythonProject :: [FilePath] -> IO (Semantic.Graph.Graph ControlFlowVertex)
-callGraphPythonProject paths = runTaskOrDie $ do
+callGraphPythonProject :: Path.RelFile -> IO (Semantic.Graph.Graph ControlFlowVertex)
+callGraphPythonProject path = runTaskOrDie $ do
```
Comment (robrix) [inline test/Graphing/Calls/Spec.hs]: So this is semantically different, but we only ever use it with single paths anyway; is that the idea?

File: test/Graphing/Calls/Spec.hs [hunk 4]
```diff
@@ -6,19 +6,19 @@ import Prelude hiding (readFile)
...
   let proxy = Proxy @'Language.Python
   let lang = Language.Python
-  blobs <- catMaybes <$> traverse readBlobFromFile (flip File lang <$> paths)
-  package <- fmap snd <$> parsePackage pythonParser (Project (takeDirectory (maybe "/" fst (uncons paths))) blobs lang [])
+  blobs <- catMaybes <$> traverse readBlobFromFile (flip File lang <$> [Path.toString path])
```
Comment (robrix) [inline test/Graphing/Calls/Spec.hs]:
```suggestion
  blobs <- catMaybes <$> traverse readBlobFromFile [File (Path.toString path) lang]
```

File: test/Examples.hs [hunk 5]
```diff
@@ -38,16 +39,17 @@ main = withOptions opts $ \ config logger statter -> hspec . parallel $ do
...
   where
     parseExamples session LanguageExample{..} tsDir = do
       knownFailures <- runIO $ knownFailuresForPath tsDir languageKnownFailuresTxt
-      files <- runIO $ globDir1 (compile ("**/*" <> languageExtension)) (tsDir </> languageExampleDir)
-      for_ files $ \file -> it file $ do
-        res <- runTask session (parseFilePath file)
+      files <- runIO $ globDir1 (compile ("**/*" <> languageExtension)) (Path.toString (tsDir </> languageExampleDir))
+      let paths = Path.relFile <$> files
+      for_ paths $ \file -> it (Path.toString file) $ do
+        res <- runTask session (parseFilePath (Path.toString file))
```
Comment (robrix) [inline test/Examples.hs]: This particular instance feels a little counterproductive, but I suppose that’ll be improved when `parseFilePath` is amended to take a typed path.

Comment (patrickt) [reply test/Examples.hs]: Oh, lemme do that now.

File: test/Graphing/Calls/Spec.hs (hunk 3 above)
Comment (patrickt) [reply test/Graphing/Calls/Spec.hs]: Yes.

File: test/Examples.hs (hunk 2 above)
Comment (patrickt) [reply test/Examples.hs]: big lol

File: test/Graphing/Calls/Spec.hs [hunk 6]
```diff
@@ -17,8 +17,8 @@ callGraphPythonProject :: Path.RelFile -> IO (Semantic.Graph.Graph ControlFlowVe
 callGraphPythonProject path = runTaskOrDie $ do
   let proxy = Proxy @'Language.Python
   let lang = Language.Python
-  blobs <- catMaybes <$> traverse readBlobFromFile (flip File lang <$> [Path.toString path])
-  package <- fmap snd <$> parsePackage pythonParser (Project (Path.toString (Path.takeDirectory path)) blobs lang [])
+  blob <- readBlobFromFile' (fileForRelPath path)
```
Comment (robrix) [inline test/Graphing/Calls/Spec.hs]: This is way better 👍 I hadn’t noticed how far the hoop-jumping extended!

**Pull request merged**